packages/babel-helper-create-class-features-plugin/src/fields.js

```diff
@@ -294,7 +294,9 @@ const privateNameHandlerSpec = {
           value,
         ]);
       }
-      return t.callExpression(file.addHelper("classPrivateMethodSet"), []);
+      // Although classPrivateMethodSet accepts no arguments and throws runtime
+      // error, `value` should always be evaluated
+      return t.callExpression(file.addHelper("classPrivateMethodSet"), [value]);
```

Comment:
Although we could further optimize the output, e.g. only plug in `value` if it is _not_ static. But I don't think it worths the efforts since a runtime error will be always thrown later.